### PR TITLE
Add Auto Author Assign Workflow for Pull Requests

### LINKED ISSUE #12603: Auto Assign PR to the Author
### What we should change and why:
- There's an amazing workflow called [Auto Author Assign](https://github.com/marketplace/actions/auto-author-assign) which automatically assigns the PR to the author.
- I believe most of the time, the one who opens the PR is the attended person to whom the PR should be assigned.
- This workflow would eliminate the need for manual assignment.

### PR DESCRIPTION
**Pull request by chahmedejaz**
#### What? Why?

- Closes #12603 
- This PR adds the workflow to assign the PR to the author when it's opened or reopened
- - After merging this, other PRs should be automatically assigned to the author.

#### What should we test?
- After merging this, other PRs should be automatically assigned to the author.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

### CODE REVIEW

File: .github/workflows/auto-author-assign.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Auto Author Assign
+
+on:
+  pull_request_target:
+    types: [opened, reopened]
```
Comment (mkllnk) [inline .github/workflows/auto-author-assign.yml:5]: I'm not sure we want the automation on reopen but let's keep it for now and see how it goes. An author should already be assigned in that case anyway.

**Review by mkllnk — APPROVED** [review]
Great. Happy to merge without further testing.

**Review by dacook — APPROVED** [review]
Looks good to me 👍 

Thanks for setting up the test to confirm this. 
As it's not application code, I think it would be ok to have just merged it without testing, because it's quite easy to undo. But I appreciate the certainty we have about it now :)

File: .github/workflows/auto-author-assign.yml (hunk 1 above)
Comment (chahmedejaz) [reply .github/workflows/auto-author-assign.yml:5]: Yes, you are right about it. The good thing about this action is if the PR is already assigned or the PR is created by the bot (like _dependabot_), then it skips the auto-assignment process. :)

**Pull request merged**